[16.0][ADD] helpdesk_product

MariaBForgeFlow opened the pull request:
This module adds the option to select product in the helpdesk tickets.

@ForgeFlow

MariaBForgeFlow committed:
[ADD] helpdesk_product

dalonsod commented on helpdesk_product/__manifest__.py:
```diff
@@ -0,0 +1,13 @@
+    "license": "AGPL-3",
+    "category": "After-Sales",
+    "website": "https://github.com/OCA/helpdesk",
+    "depends": ["helpdesk_mgmt", "sale_management"],
```
Is needed such dependency, shoudn't be `product` addon instead enough?

dalonsod commented on helpdesk_product/views/helpdesk_ticket_view.xml:
```diff
@@ -0,0 +1,41 @@
+                <field name="user_id" position="after">
+                    <field name="product_id" />
+                </field>
+            </field>
```
You could add some shortcuts:
* Filter for tickets with product.
* Group by product

dalonsod reviewed (changes requested):
Hello @mariadforgeflow thanks for the contribution!

See comments, and also and not blocking, I'd suggest adding a product flag (in template) and/or category that indicates whether a product is available for tickets, so some product could be excluded for this functionality if needed.

MariaBForgeFlow committed:
[ADD] helpdesk_product

MariaBForgeFlow committed:
[ADD] helpdesk_product

MariaBForgeFlow committed:
Merge branch '16.0-add-helpdesk_product' of github.com:ForgeFlow/helpdesk into 16.0-add-helpdesk_product

MariaBForgeFlow committed:
Merge branch '16.0-add-helpdesk_product' of github.com:ForgeFlow/helpdesk into 16.0-add-helpdesk_product

MariaBForgeFlow committed:
Merge branch '16.0-add-helpdesk_product' of github.com:ForgeFlow/helpdesk into 16.0-add-helpdesk_product

MariaBForgeFlow commented:
Hello @dalonsod 

I have modified the module to include the following :

- Option to make certain products unavailable for tickets
- Filter for tickets which contain associated products
- Group tickets by product

I have also changed the module's dependencies so that it depends on `"product"` instead of `"sale_management"`.

dalonsod commented on helpdesk_product/views/helpdesk_ticket_view.xml:
```diff
@@ -0,0 +1,55 @@
+                    <filter
+                    string="Product Related"
+                    name="product_filter"
+                    domain="['!', ('product_id', '!=', '' )]"
```
Simpler domain IMO:
```suggestion
                    domain="[('product_id', '!=', False)]"
```

dalonsod commented on helpdesk_product/views/helpdesk_ticket_view.xml:
```diff
@@ -0,0 +1,55 @@
+            <field name="inherit_id" ref="helpdesk_mgmt.ticket_view_tree" />
+            <field name="arch" type="xml">
+                <field name="user_id" position="after">
+                    <field name="product_id" />
```
Make it optional?

dalonsod reviewed (approved):
Tested on runboat :+1: 
See minor comments anyway. And I only miss row things: a "Products enabled for helpdesk" menu for Helpdesk Managers, like e.g. e-commerce enabled products, and some demo data, but they're not blocking IMO.

MariaBForgeFlow committed:
[ADD] helpdesk_product: small fixes

MariaBForgeFlow committed:
[ADD] helpdesk_product: add demo data

MariaBForgeFlow commented:
> Tested on runboat 👍 See minor comments anyway. And I only miss row things: a "Products enabled for helpdesk" menu for Helpdesk Managers, like e.g. e-commerce enabled products, and some demo data, but they're not blocking IMO.

Hello David,

I don't quite understand what you mean by having a "Products enabled for helpdesk" menu. Do you mean adding a filter on the product search view to search by products that are enabled for helpdesk?

dalonsod commented:
I mean something like this:

![imagen](https://github.com/OCA/helpdesk/assets/12749832/e3e4b7d3-2aa5-4e54-a247-aff67208e5e2)

Adding a shortcut to products management within Helpdesk app like e-commerce does.

MariaBForgeFlow commented:
I will look into it, thank you.

MariaBForgeFlow committed:
[ADD] helpdesk_product: add products menu to Helpdesk app

MariaBForgeFlow commented:
Hello @dalonsod 

I added a menu to the products that are available for helpdesk tickets on the Helpdesk application so that they can be accessed from Helpdesk. Please let me know if there's any other functionality you find missing.

dalonsod commented:
:+1:

DavidJForgeFlow reviewed (approved):
LGTM!

flotho reviewed (commented):
Thanks for the contributoin.

Regarding the technical everything looks fine but I didn't understood the use case behing this module. 
Any chance to have an explanation of the business you're expecting with this module ?
Regards

MariaBForgeFlow commented:
> Thanks for the contributoin.
> 
> Regarding the technical everything looks fine but I didn't understood the use case behing this module. Any chance to have an explanation of the business you're expecting with this module ? Regards

Hello Florent,

This module will allow helpdesk users to link a product on a helpdesk ticket to the product's record. One example of how this can be useful is for retail companies, as they might receive complaints from customers regarding specific products, and linking the product on the helpdesk ticket will provide more information to the helpdesk user.
Please let me know if there's anything else I can clarify.

JordiBForgeFlow commented:
@dalonsod Can you merge this one?

dalonsod commented:
I'm afraid I can't. I have not such access level in this repo 😅

DarioLodeiros commented:
/ocabot merge patch

rousseldenis commented:
@JordiBForgeFlow @LoisRForgeFlow Could you confirm this module is to select the product concerned by the helpdesk ticket (and not a 'Helpdesk Product') ? Thanks

LoisRForgeFlow commented:
@rousseldenis I was not involved in this one but looking at this comment https://github.com/OCA/helpdesk/pull/520#issuecomment-2036757249, I would say is the the product concerned by the helpdesk ticket.

rousseldenis commented:
> @rousseldenis I was not involved in this one but looking at this comment [#520 (comment)](https://github.com/OCA/helpdesk/pull/520#issuecomment-2036757249), I would say is the the product concerned by the helpdesk ticket.

@LoisRForgeFlow Thanks!
